File: benchmark/lua/mean_variance.lua
local start = os.clock()
local arr = {}
for i = 1, 1000000 do arr[i] = i end
local s = 0
for i = 1, 1000000 do s = s + arr[i] end
local mean = s / #arr
local sq_diff = 0
for i = 1, #arr do
    local diff = arr[i] - mean
    sq_diff = sq_diff + diff * diff
end
local variance = sq_diff / #arr
local finish = os.clock()
print("Mean: " .. mean)
print("Variance: " .. variance)
print(string.format("Time: %.0fms", (finish - start) * 1000))

File: benchmark/src/ruby/mean_variance.rb
arr = (1..1000000).to_a
s = 0
start = Time.now.to_f * 1000
arr.each { |x| s += x }
mean = s.to_f / arr.length
sq_diff = 0
arr.each do |x|
    d = x - mean
    sq_diff += d * d
end
variance = sq_diff / arr.length
finish = Time.now.to_f * 1000
puts "Result: #{mean.round(1)}/#{variance.round(1)}"
puts "Time: #{(finish - start).round}ms"

